### PR TITLE
Added Shellcheck

### PR DESCRIPTION
**Pull request by mustaphazorgati**
Hey @rokroskar,

since I don't have write access to this repository I push any commit for #24.

That's why I created this PR. This closes #24.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -35,28 +35,28 @@ function convertToKeyValuePairs() {
...
 }
 
 # extract meta information for current workflow run
 exportAll "$(curl -s "${GITHUB_API}/repos/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}" -H "${auth_header}" -H "${accept_header}" | extractMetaInformation | convertToKeyValuePairs)"
-echo "workflow id: $workflow_id"
-echo "branch: $branch"
-echo "repo: $repo"
+echo "workflow id: ${workflow_id?}"
+echo "branch: ${branch?}"
+echo "repo: ${repo?}"
```
Comment (mustaphazorgati) [inline entrypoint.sh:51]: According to shellcheck this way of creating variables is not good. The error message is that we use variables which are not declared. Technically speaking they are not declared at compile time, but will be declared at run time by the `exportAll` function.

Currently we use the function `convertToKeyValuePairs` which resolves the json object generated by `extractMetaInformation` to list of valid export strings. That list is then exported as variables.

Do we want to keep that behavior? I admit that this implicit way of declaring the variables can be a little confusing. Making it explicit could increase the readability at the cost of code compactness. What do you thing?

As of right now I just muted all error messages concerning this issue. If we want to change the way we generate our variables we should do that as a separate issue.

Comment (rokroskar) [reply entrypoint.sh:51]: Hi @mustaphazorgati yes, I wondered as well how we should fix this - I think code readability in a script like this is more important than compactness so I would vote for the more verbose approach and not muting this particular error.

Comment (mustaphazorgati) [reply entrypoint.sh:51]: -> #26

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -1,7 +1,11 @@
+name: 'CI'
+
 on:
+  workflow_dispatch:
```
Comment (rokroskar) [inline .github/workflows/ci.yaml]: why is this needed?

Comment (mustaphazorgati) [reply .github/workflows/ci.yaml]: This is a quality of life thing. It allows you (and other people who have write access to this repo) to manually start the workflow. Can be useful sometimes.

Comment (mustaphazorgati) [reply .github/workflows/ci.yaml]: See here: https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow#running-a-workflow-on-github

Comment (rokroskar) [reply .github/workflows/ci.yaml]: It can be, but I find that if I have a pretty loose CI policy like trigger on every push, I don't really ever have a need for this. I'd prefer to leave it as it was in #24.

Comment (mustaphazorgati) [reply .github/workflows/ci.yaml]: removed. :)

**Review by rokroskar — APPROVED** [review]
Thanks @mustaphazorgati!

**Comment by mustaphazorgati** [comment]
I will rebase the other prs shortly.
